### PR TITLE
WPB-5098 Backend-to-backend OpenApi Docs

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/WPB-5098

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: charts/nginz/values.yaml [hunk 1]
```diff
@@ -162,6 +162,10 @@ nginx_conf:
       disable_zauth: true
       envs:
       - staging
+    - path: /api-federation/swagger-ui
```
Comment (fisx) [inline charts/nginz/values.yaml:165]: can you update https://docs.wire.com/understand/api-client-perspective/swagger.html to mention this?

File: libs/wire-api-federation/src/Wire/API/Federation/Domain.hs [hunk 2]
```diff
@@ -58,3 +59,6 @@ instance
 
 originDomainHeaderName :: IsString a => a
 originDomainHeaderName = fromString $ symbolVal (Proxy @OriginDomainHeaderName)
+
+instance (HasOpenApi api) => HasOpenApi (OriginDomainHeader :> api) where
```
Comment (fisx) [inline libs/wire-api-federation/src/Wire/API/Federation/Domain.hs:63]: i'm not sure, but shouldn't `OriginDomainHeader` also be documented somehow?  If in doubt, add something to the description?

File: libs/wire-api/src/Wire/API/Conversation/Action.hs [hunk 3]
```diff
@@ -89,6 +89,40 @@ instance ToJSON SomeConversationAction where
         actionJSON = fromMaybe A.Null $ schemaOut (conversationActionSchema sb) action
      in A.object ["tag" A..= tag, "action" A..= actionJSON]
 
+instance S.ToSchema SomeConversationAction where
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Conversation/Action.hs:92]: i guess it's ok to not use schema-profunctor, but then can you add a roundtrip- and/or unit-test?

**Review by fisx — APPROVED** [review]

File: libs/wire-api/src/Wire/API/Conversation/Action.hs (hunk 3 above)
Comment (battermann) [reply libs/wire-api/src/Wire/API/Conversation/Action.hs:92]: Roundtrip tests for the `ToJSON` and `FromJSON` instances, which I haven't touched, already exist. The `ToSchema` instance is just for the API docs.

**Pull request merged**